### PR TITLE
[JENKINS-51316] Remove generateKey functionality

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-51316](https://issues.jenkins-ci.org/browse/JENKINS-51316) for the reasons behind this change.

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/EC2Cloud.java [hunk 1]
```diff
@@ -941,7 +941,7 @@ protected FormValidation doTestConnection(URL ec2endpoint, boolean useInstancePr
                 ec2.describeInstances();
 
                 if (privateKey == null)
-                    return FormValidation.error("Private key is not specified. Click 'Generate Key' to generate one.");
+                    return FormValidation.error("Private key is not specified. Please fill up the private key field with a valid one.");
```
Comment (jvz) [inline src/main/java/hudson/plugins/ec2/EC2Cloud.java]:
```suggestion
                    return FormValidation.error("Private key is not specified. Please fill the private key field with a valid one.");
```

File: README.md [hunk 2]
```diff
@@ -28,11 +28,8 @@ UI for configuring the EC2 plugin.  Then enter the Access Key and Secret
 Access Key which act like a username/password (see IAM section). Because
 of the way EC2 works, you also need to have an RSA private key that the
 cloud has the other half for, to permit sshing into the instances that
-are started. If you have already been using EC2 and have your own key,
-you can paste it here. Otherwise, you can have Jenkins generate one. If
-you let Jenkins generate one, you should save this private key in your
-file system as well, as you'll need this to interactively logon to EC2
-instances.
++are started. Please use the AWS console or any other tool of your choice
++ to generate the private key to interactively logon to EC2 instances.
```
Comment (jvz) [inline README.md]:
```suggestion
+ to generate the private key to interactively log in to EC2 instances.
```

File: README.md [hunk 3]
```diff
@@ -28,11 +28,8 @@ UI for configuring the EC2 plugin.  Then enter the Access Key and Secret
 Access Key which act like a username/password (see IAM section). Because
 of the way EC2 works, you also need to have an RSA private key that the
 cloud has the other half for, to permit sshing into the instances that
-are started. If you have already been using EC2 and have your own key,
-you can paste it here. Otherwise, you can have Jenkins generate one. If
-you let Jenkins generate one, you should save this private key in your
-file system as well, as you'll need this to interactively logon to EC2
-instances.
++are started. Please use the AWS console or any other tool of your choice
```
Comment (fcojfernandez) [inline README.md]: extra `+` character?
```suggestion
are started. Please use the AWS console or any other tool of your choice
```

File: README.md [hunk 4]
```diff
@@ -28,11 +28,8 @@ UI for configuring the EC2 plugin.  Then enter the Access Key and Secret
...
++ to generate the private key to interactively log in to EC2 instances.
```
Comment (fcojfernandez) [inline README.md]: ditto
```suggestion
to generate the private key to interactively log in to EC2 instances.
```

**Comment by fcojfernandez** [comment]
@res0nance @jvz is fine for you if we release a new version containing this change and hence without a non-working feature?

**Comment by jvz** [comment]
Fine with me.

**Comment by res0nance** [comment]
Perfectly alright with me

**Comment by fcojfernandez** [comment]
I will release it on Monday morning then! thanks everyone

**Pull request merged**